### PR TITLE
Fixes reexports in search

### LINKED ISSUE #96681: doc aliases don't work on reexports
For example with this code:

```rust
#[doc(alias = "->a")]
pub use ::std as x;
#[doc(alias = "->b")]
pub struct Foo;
```

We can't search for `->a` but we can search for `->b`.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #96681.

At some point we stopped reexporting items in search so this PR fixes it.

It also adds a regression test.

r? @notriddle

### CODE REVIEW

**Review by notriddle — CHANGES_REQUESTED** [review]
Simple nit in an otherwise good PR.

If there's no reason not to use `if let` chaining, switch over, then r=me.

File: src/librustdoc/formats/cache.rs [hunk 1]
```diff
@@ -248,7 +248,13 @@ impl<'a, 'tcx> DocFolder for CacheBuilder<'a, 'tcx> {
...
 
         // Index this method for searching later on.
-        if let Some(ref s) = item.name {
+        if let Some(ref s) = item.name.or_else(|| {
+            if let clean::ImportItem(ref i) = *item.kind {
+                if let clean::ImportKind::Simple(s) = i.kind { Some(s) } else { None }
+            } else {
+                None
+            }
+        }) {
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustdoc/formats/cache.rs:257]: That's great! This feature is really nice. :)

**Comment by GuillaumeGomez** [comment]
My PR actually had another bug: the sidebar wasn't listing `Re-exports` even if there was some on the page. Same bug, fixed the same way.

**Comment by GuillaumeGomez** [comment]
And fixed. :)

**Review by notriddle — APPROVED** [review]

**Pull request merged**